lib/lorekeeper/fast_logger.rb

```diff
@@ -56,10 +56,13 @@ def initialize(file)
     # This is part of the standard Logger API, we need this to be compatible
     def add(severity, message_param = nil, progname = nil, &block)
       return true if severity < @level
-      message = (block && block.call) || message_param || progname
+
+      message = block&.call || message_param || progname
       log_data(severity, message.freeze)
     end
 
+    # rubocop:disable Lint/UnusedMethodArgument
```

Comment:
instead of disabling the `Lint/UnusedMethodArgument` cop, how about checking `block` itself?
https://msp-greg.github.io/rubocop-performance/RuboCop/Cop/Performance/BlockGivenWithExplicitBlock.html
```diff
   def silence_logger(&block)
-     yield if block_given?
+     yield if block
    end
```